internal/backup.go

```diff
@@ -107,48 +108,49 @@ func UploadDto(folder storage.Folder, dto interface{}, path string) error {
 	return folder.PutObject(path, r)
 }
 
-func (backup *Backup) CheckExistence() (bool, error) {
-	exists, err := backup.SentinelExists()
+func (backup *Backup) CheckExistence() (exists bool, storage string, err error) {
+	exists, storage, err = backup.SentinelExists()
```

Comment:
Backup is logically defined by name + location (storage). Instead of passing the backup storage name from all these functions, let's make it a field of the `Backup` struct itself.